minari/utils.py

```diff
@@ -172,7 +172,8 @@ def validate_datasets_to_combine(datasets_to_combine: List[MinariDataset]) -> En
         max_episode_steps = None
     else:
         max_episode_steps = max(
-            dataset.spec.env_spec.max_episode_steps for dataset in datasets_to_combine
+            dataset.spec.env_spec.max_episode_steps
+            for dataset in datasets_to_combine  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
Actually, since now the datasets can have a env_spec=None, we should address also this case.
You can create the list of env_spec and filter out None values

Reply:
As of now, I'm raising an error if any of the datasets to be combined have no env_spec. Please do let me know if you'd prefer some other specific action instead of this.